zebra/if_netlink.c

```diff
@@ -98,10 +99,26 @@ static void set_ifindex(struct interface *ifp, ifindex_t ifi_index,
 					EC_LIB_INTERFACE,
 					"interface rename detected on up interface: index %d was renamed from %s to %s, results are uncertain!",
 					ifi_index, oifp->name, ifp->name);
-			if_delete_update(oifp);
+			update = true;
```

Comment:
If we don't do `if_delete_update` here, then the call to `if_set_index` a couple of lines later won't work as expected, because of the ifindex collision.